docs: update nodescan rfc (#1207)
rancher-sandbox/sbombastic: docs/rfc/0008_node_scan.md

## Scan Workflow

> 1. The user applies a `NodeScanConfiguration` with a defined `scanInterval`. Optionally, the user can annotate the resource with `sbomscanner.kubewarden.io/node-rescan-requested: "true"` to trigger an immediate scan.

Reviewer: we need to make clear that this is triggering the scan for all nodes

Reviewer: @alegrey91 this was not addressed

2. The controller creates a `NodeScanJob` for each node matching the `nodeSelector` (or all nodes if no selector is specified).
3. Each worker subscribes to the NATS subjects `sbomscanner.nodesbom.generate.{node-name}` and `sbomscanner.nodesbom.scan.{node-name}`, and receives the scan job for its node.
4. The worker executes the scan, generating a `NodeSBOM` and a `NodeVulnerabilityReport` for the node.
5. The results are stored in the cluster and can be accessed by the user for review and remediation.

To let users easily understand the flow, here's a simple diagram:

![NodeScan Workflow](./assets/nodescan-wf.png)

Without the `NodeScanConfiguration`, users cannot apply a `NodeScanJob` independently,
since the `NodeScanConfiguration` holds the configuration used by all `NodeScanJob` resources
(e.g., `skipPatterns`, `nodeSelector`, `platforms`).

> When a new `NodeScanJob` would be created, the runner checks if another `NodeScanJob` is already in progress for the same node.

Reviewer: ```suggestion
When a new `NodeScanJob` is about to be created, the runner checks whether another `NodeScanJob` is already in progress for the same node.
```

> If there is an active job, the new job creation is skipped entirely.

Reviewer: ```suggestion
If another job is already active, the new job is not created.
```
